### ISSUE TITLE
Flair Collapse: NameError: name 'res' is not defined

### ISSUE THREAD

**Issue by itslittman** [BrooksLabUCSC/flair#215]
I can successfully run Flair Correct, but not Collapse, as I get the error: NameError: name 'res' is not defined.

Someone else reported this error too (https://github.com/BrooksLabUCSC/flair/issues/174), but the only solution listed was downgrading Python to 3.7 and that did not work for me.

**Comment by Jeltje** [comment]
I have just made a commit of the code that generates this error, `collapse_isoforms_precise.py`, what happens if you try that version? You can either download and replace that one file under flair/src/flair/, or `git clone` the current master branch.

If you still get the error, or something similar, please attach a sample of your input file and give us the full command you are trying to run.

Thanks!

**Comment by itslittman** [comment]
Hey thanks for the reply!! We're getting somewhere. I replaced the script, and that error is gone, but now I get:
[image0.jpeg]

Sent from my iPhone

On Aug 11, 2022, at 5:19 PM, Jeltje ***@***.***> wrote:

﻿

I have just made a commit of the code that generates this error, collapse_isoforms_precise.py, what happens if you try that version? You can either download and replace that one file under flair/src/flair/, or git clone the current master branch.

If you still get the error, or something similar, please attach a sample of your input file and give us the full command you are trying to run.

Thanks!

**Comment by Jeltje** [comment]
I think your previous post was supposed to have an image link or attachment, but all I see is [image0.jpeg]
Can you try again?

**Comment by itslittman** [comment]
oh my bad:
[image0.jpeg]

Sent from my iPhone

On Aug 11, 2022, at 8:47 PM, Jeltje ***@***.***> wrote:

﻿

I think your previous post was supposed to have an image link or attachment, but all I see is [image0.jpeg]
Can you try again?

**Comment by itslittman** [comment]
if it doesn't load again, it said:

Writing temporary files to /var/folders/y9/6jbrb7w96wb0r6jywf6f_3f80000gn/T/tmp3n2afklz/
Annotated ends extracted from GTF
Read data extracted
[Annotated ends extracted from GTF
Annotated ends extracted from GTF
Annotated ends extracted from GTF
Annotated ends extracted from GTF
Read data extracted
Read data extracted
Single-exon genes grouped, collapsing
Single-exon genes grouped, collapsing
[multiprocessing.pool.RemoteTraceback:

and then gave the traceback:

Traceback (most recent call last):
File
"/Users/noahlittman/miniconda/lib/python3.10/multiprocessing/pool.py", line 125, in worker
result = (True, func (*args, **kwds))
File "/Users/noahlittman/miniconda/lib/python3.10/multiprocessing/pool.py", line 48 in mapstar
return list (map(*args))
File "/Users/noahlittman/miniconda/lib/python3.10/site-packages/flair/collapse_isoforms_precise.py",line233,inrun_i
terative add se
se ordered
= all_se_by_chrom[chrom].keys()
TypeError:
'NoneType!
object is not subscriptable
atio
nts tr
The above exception was the direct cause of the following exception:
Traceback (most recent call last) :
File
"/Users/noahlittman/miniconda/lib/python3.10/site-packages/flair/collapse_isoforms_precise.py",line674,in
<modu
le>
res
= p.map (run_iterative_add_se,
, chrom_names)
File "/Users/noahlittman/miniconda/lib/python3.10/multiprocessing/pool.py", line 364, in map
return self. _map _async(func, iterable, mapstar, chunksize).get ()
File "/Users/noahlittman/miniconda/lib/python3.10/multiprocessing/pool.py", line 771, in get
raise self. value


And the official  error message was:

TypeError: 'NoneType' object is not subscriptable

Sent from my iPhone

On Aug 11, 2022, at 8:55 PM, Littman, Noah ***@***.***> wrote:

﻿ oh my bad:
[image0.jpeg]

Sent from my iPhone

On Aug 11, 2022, at 8:47 PM, Jeltje ***@***.***> wrote:

﻿

I think your previous post was supposed to have an image link or attachment, but all I see is [image0.jpeg]
Can you try again?

**Comment by Jeltje** [comment]
The image still doesn't work, but the plain text is great, thanks.

Would you mind sending me your input psl or bed file (or just the first ten lines of it)? jeltje@soe.ucsc.edu
I can't reproduce the error with my own test input.
Also please give me the **full command** you are using.

**Comment by itslittman** [comment]
It's all fixed now!! 
I combed through the online repository again after repeatedly encountering the second error, and realized that the first yaml file in /misc was removed and marked as bad. I cloned the whole repository again/used the other yaml file, and did a fresh install and everything works! It does make you update samtools, so I just copied and pasted my newer samtools from /opt/local/bin as a quick fix. 

Thank you for the help and that lightning-fast debug of the first script!

Best,
Noah
